src/sentry/api/event_search.py

```diff
@@ -597,39 +604,45 @@ def visit_numeric_filter(self, node, children):
         return self._handle_numeric_filter(search_key, operator, search_value)
 
-    def visit_aggregate_filter(self, node, children):
+    def visit_aggregate_duration_filter(self, node, children):
         (negation, search_key, _, operator, search_value) = children
         operator = handle_negation(negation, operator)
-        search_value = search_value[0]
 
         try:
-            aggregate_value = None
-            value_type = search_value[0]
-
-            if value_type in ["duration_format", "percentage_format"]:
-                # Even if the search value matches duration format, only act as duration for certain columns
-                function = resolve_field(
-                    search_key.name, self.params, functions_acl=FUNCTIONS.keys()
-                )
-                if function.aggregate is not None:
-                    if value_type == "percentage_format" and self.is_percentage_key(
-                        function.aggregate[0]
-                    ):
-                        aggregate_value = parse_percentage(search_value[1])
-                    # Extract column and function name out so we can check if we should parse as duration
-                    elif value_type == "duration_format" and self.is_duration_key(
-                        function.aggregate[1]
-                    ):
-                        aggregate_value = parse_duration(*search_value[1:])
-
-            if aggregate_value is None:
-                # Duration can fall through as a numeric value due to the
-                # overlapping 'm'
-                if len(search_value) == 3:
-                    search_value = search_value[1:]
+            # Even if the search value matches duration format, only act as
+            # duration for certain columns
+            function = resolve_field(search_key.name, self.params, functions_acl=FUNCTIONS.keys())
+            if function.aggregate is not None and self.is_duration_key(function.aggregate[1]):
+                aggregate_value = parse_duration(*search_value)
+            else:
+                # Duration overlaps with numeric values
                 aggregate_value = parse_numeric_value(*search_value)
-        except ValueError:
-            raise InvalidSearchQuery(f"Invalid aggregate query condition: {search_key}")
+        except InvalidQuery as exc:
+            raise InvalidSearchQuery(str(exc))
+
+        return AggregateFilter(search_key, operator, SearchValue(aggregate_value))
+
+    def visit_aggregate_percentage_filter(self, node, children):
+        (negation, search_key, _, operator, search_value) = children
+        operator = handle_negation(negation, operator)
+
+        try:
+            function = resolve_field(search_key.name, self.params, functions_acl=FUNCTIONS.keys())
+            if function.aggregate is not None and self.is_percentage_key(function.aggregate[0]):
+                aggregate_value = parse_percentage(search_value)
+            else:
```

Comment:
nit: can we dupe the 
`Duration overlaps with numeric values`
comment here too?

Reply:
yep